judge/views/problem.py

```diff
@@ -565,6 +565,9 @@ def problem_submit(request, problem=None, submission=None):
             else:
                 model = form.save()
 
+            # Create the SubmissionSource object
+            source = SubmissionSource(submission=model, source=form.cleaned_data['source'])
+            source.save()
```

Comment:
You should consider `model.source = source` so that you save a query in `judge_submission`.